scripts/helm-validate.py

```diff
@@ -0,0 +1,219 @@
+                        chart_repository, parent_contents
+                    )
+
+                self.logger.debug(
+                    f"Name: {chart_name} Version: {chart_version} Repository: {chart_repository}"
+                )
+
+                self.charts[chart_name] = {
+                    "version": chart_version,
+                    "repository": chart_repository,
+                }
+
+        if not self.charts:
+            self.logger.debug("Could not find any helm_release under module resources")
+
+        return self.charts
+
+    def generate_helm_chart_index(self):
+        """
+        Generate an index of helm charts by searching for helm_release resources in Terraform files.
+
+        Returns:
+            A dictionary where the keys are the names of the charts and the values are dictionaries containing the chart's
+            version and repository.
+
+        Raises:
+            ValueError: If no helm charts are found in the Terraform files.
+        """
+        paths = self.get_filepaths_that_contain_helm_release()
+        helm_charts = {}
+        for path in paths:
+            helm_information = self.retrieve_helm_information(path)
+            helm_charts.update(helm_information)
+
+        if not helm_charts:
+            raise ValueError("No helm charts found in the Terraform files.")
+
+        with open("helm_charts.json", "w") as f:
+            json.dump(helm_charts, f)
+
+        return helm_charts
+
+
+def pull_helm_chart(chart_index: dict, skip_charts: list) -> None:
+    """
+    Pull helm charts specified in `chart_index` and save them in the `helm_charts` directory.
+
+    Args:
+        chart_index: A dictionary containing chart names as keys and chart metadata (version and repository)
+            as values.
+        skip_charts: A list of chart names to skip.
+
+    Raises:
+        ValueError: If a chart could not be found in the `helm_charts` directory after pulling.
+    """
+    chart_dir = "helm_charts"
+    os.makedirs(chart_dir, exist_ok=True)
+    os.chdir(chart_dir)
+
+    for chart_name, chart_metadata in tqdm(
+        chart_index.items(), desc="Downloading charts"
+    ):
+        chart_version = chart_metadata["version"]
+        chart_repository = chart_metadata["repository"]
+
+        if chart_name in skip_charts:
+            continue
+
+        os.system(f"helm repo add {chart_name} {chart_repository}")
+        os.system(
+            f"helm pull {chart_name} --version {chart_version} --repo {chart_repository} --untar"
+        )
+
+        chart_filename = f"{chart_name}-{chart_version}.tgz"
+        if not os.path.exists(chart_filename):
+            raise ValueError(
+                f"Could not find {chart_name}:{chart_version} directory in {chart_dir}."
+            )
+
+    print("All charts downloaded successfully!")
+    # shutil.rmtree(pathlib.Path(os.getcwd()).parent / chart_dir)
+
+
+def add_workflow_job_summary(chart_index: dict):
```

Comment:
Could this be generalized so that we can run this script locally?

Reply:
I moved this out of the regular PRs/pushes (unless it touches the workflow file itself) and added a manual trigger as well as made it a  CRON job instead to avoid in the future this becoming a blocker for new contributors